Conjunto de dados com de mais de 1.000 registros, contendo informações sobre produtos relativos ao e-commerce da empresa Amazon. O mesmo é proveniente de terceiros, adquirido por meio da plataforma [Kaggle](https://www.kaggle.com/datasets/karkavelrajaj/amazon-sales-dataset).  

**Sobre o projeto**  
A empresa Amazon.com, Inc. é uma empresa multinacional de tecnologia norte-americana com sede em Seattle, Washington. A companhia trabalha na área do e-commerce, computação em nuvem, streaming e inteligência artificial.  
Neste projeto, ao discorrer sobre dados inerentes ao e-commerce da empresa, o observador terá esclarecimento acerca de algumas questões, relativas a compras no âmbito de loja virtual da Amazon. Ao final, serão respondidas algumas perguntas de negócio.

# **PREPARAÇÃO, ORGANIZAÇÃO E ESTRUTURAÇÃO DOS DADOS**
Para a mabipulação e análise de dados, será utilizada a biblioteca pandas.

In [1]:
import pandas as pd

In [2]:
pd.set_option("display.max_columns", 30) # altera o número máximo de colunas a serem exibidas

In [3]:
df = pd.read_csv("/content/drive/MyDrive/projetos/amazon/amazon.csv")
df

product_id                                       product_name  \
0     B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1     B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2     B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3     B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4     B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   
...          ...                                                ...   
1460  B08L7J3T31  Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...   
1461  B01M6453MB  Prestige Delight PRWO Electric Rice Cooker (1 ...   
1462  B009P2LIL4  Bajaj Majesty RX10 2000 Watts Heat Convector R...   
1463  B00J5DYCCA  Havells Ventil Air DSP 230mm Exhaust Fan (Pist...   
1464  B01486F4G6  Borosil Jumbo 1000-Watt Grill Sandwich Maker (...   

                                               category discounted_price  \
0     Computers&Accessories|Accessories&Peripherals|...             ₹399   
1     Computers&Accessories|Accessories&Peripherals|...             ₹199   
2     Computers&Accessories|Accessories&Peripherals|...             ₹199   
3     Computers&Accessories|Accessories&Peripherals|...             ₹329   
4     Computers&Accessories|Accessories&Peripherals|...             ₹154   
...                                                 ...              ...   
1460  Home&Kitchen|Kitchen&HomeAppliances|WaterPurif...             ₹379   
1461  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...           ₹2,280   
1462  Home&Kitchen|Heating,Cooling&AirQuality|RoomHe...           ₹2,219   
1463  Home&Kitchen|Heating,Cooling&AirQuality|Fans|E...           ₹1,399   
1464  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...           ₹2,863   

     actual_price discount_percentage rating rating_count  \
0          ₹1,099                 64%    4.2       24,269   
1            ₹349                 43%    4.0       43,994   
2          ₹1,899                 90%    3.9        7,928   
3            ₹699                 53%    4.2       94,363   
4            ₹399                 61%    4.2       16,905   
...           ...                 ...    ...          ...   
1460         ₹919                 59%      4        1,090   
1461       ₹3,045                 25%    4.1        4,118   
1462       ₹3,080                 28%    3.6          468   
1463       ₹1,890                 26%      4        8,031   
1464       ₹3,690                 22%    4.3        6,987   

                                          about_product  \
0     High Compatibility : Compatible With iPhone 12...   
1     Compatible with all Type C enabled devices, be...   
2     【 Fast Charger& Data Sync】-With built-in safet...   
3     The boAt Deuce USB 300 2 in 1 cable is compati...   
4     [CHARGE & SYNC FUNCTION]- This cable comes wit...   
...                                                 ...   
1460  SUPREME QUALITY 90 GRAM 3 LAYER THIK PP SPUN F...   
1461                       230 Volts, 400 watts, 1 Year   
1462  International design and styling|Two heat sett...   
1463  Fan sweep area: 230 MM ; Noise level: (40 - 45...   
1464  Brand-Borosil, Specification – 23V ~ 5Hz;1 W...   

                                                user_id  \
0     AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1     AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2     AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3     AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4     AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   
...                                                 ...   
1460  AHITFY6AHALOFOHOZEOC6XBP4FEA,AFRABBODZJZQB6Z4U...   
1461  AFG5FM3NEMOL6BNFRV2NK5FNJCHQ,AGEINTRN6Z563RMLH...   
1462  AGVPWCMAHYQWJOQKMUJN4DW3KM5Q,AF4Q3E66MY4SR7YQZ...   
1463  AF2JQCLSCY3QJATWUNNHUSVUPNQQ,AFDMLUXC5LS5RXDJS...   
1464  AFGW5PT3R6ZAVQR4Y5MWVAKBZAYA,AG7QNJ2SCS5VS5VYY...   

                                              user_name  \
0     Manav,Adarsh gupta,Sunde

In [4]:
df.shape

(1465, 16)

Inicialmente, percebe-se o conjunto de dados com uma totalidade de 16 variáveis e 1465 registros.

## Extração e Limpeza de Dados

### Removendo valores nulos

In [5]:
df.isnull().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           2
about_product          0
user_id                0
user_name              0
review_id              0
review_title           0
review_content         0
img_link               0
product_link           0
dtype: int64

Verificando todos os registros, através das variáveis, percebe-se dois valores nulos em `rating_count`.
Os registros contendo os valores nulos (NaN) não serão removidos, tendo em vista que a variável em questão será amplamente utilizada. Assim, estes registros nulos serão posteriormente tratados.

## Renomeando Variáveis e Registros

### Renomeando registros
Os registros da variável `rating_count` contendo valores nulos serão substituídos pelo valor 0 (zero).

In [6]:
df[df.isnull().any(axis=1)]

product_id                                       product_name  \
282  B0B94JPY2N  Amazon Brand - Solimo 65W Fast Charging Braide...   
324  B0BQRJ3C47  REDTECH USB-C to Lightning Cable 3.3FT, [Apple...   

                                              category discounted_price  \
282  Computers&Accessories|Accessories&Peripherals|...             ₹199   
324  Computers&Accessories|Accessories&Peripherals|...             ₹249   

    actual_price discount_percentage rating rating_count  \
282         ₹999                 80%    3.0          NaN   
324         ₹999                 75%    5.0          NaN   

                                         about_product  \
282  USB C to C Cable: This cable has type C connec...   
324  💎[The Fastest Charge] - This iPhone USB C cabl...   

                          user_id    user_name      review_id  \
282  AE7CFHY23VAJT2FI4NZKKP6GS2UQ       Pranav   RUB7U91HVZ30   
324  AGJC5O5H5BBXWUV7WRIEIOOR3TVQ  Abdul Gafur  RQXD5SAMMPC6L   

                                     review_title  \
282  The cable works but is not 65W as advertised   
324                               Awesome Product   

                                        review_content  \
282  I have a pd supported car charger and I bought...   
324  Quick delivery.Awesome ProductPacking was good...   

                                              img_link  \
282  https://m.media-amazon.com/images/W/WEBP_40237...   
324  https://m.media-amazon.com/images/I/31-q0xhaTA...   

                                          product_link  
282  https://www.amazon.in/Amazon-Brand-Charging-Su...  
324  https://www.amazon.in/REDTECH-Lightning-Certif...

In [7]:
df.rating_count.fillna("0", inplace=True)

In [8]:
df.isnull().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           0
about_product          0
user_id                0
user_name              0
review_id              0
review_title           0
review_content         0
img_link               0
product_link           0
dtype: int64

Com isso, os valores nulos foram substituídos . Podemos verificar essa substituição através do método loc.

In [9]:
df.loc[[282, 324]]

product_id                                       product_name  \
282  B0B94JPY2N  Amazon Brand - Solimo 65W Fast Charging Braide...   
324  B0BQRJ3C47  REDTECH USB-C to Lightning Cable 3.3FT, [Apple...   

                                              category discounted_price  \
282  Computers&Accessories|Accessories&Peripherals|...             ₹199   
324  Computers&Accessories|Accessories&Peripherals|...             ₹249   

    actual_price discount_percentage rating rating_count  \
282         ₹999                 80%    3.0            0   
324         ₹999                 75%    5.0            0   

                                         about_product  \
282  USB C to C Cable: This cable has type C connec...   
324  💎[The Fastest Charge] - This iPhone USB C cabl...   

                          user_id    user_name      review_id  \
282  AE7CFHY23VAJT2FI4NZKKP6GS2UQ       Pranav   RUB7U91HVZ30   
324  AGJC5O5H5BBXWUV7WRIEIOOR3TVQ  Abdul Gafur  RQXD5SAMMPC6L   

                                     review_title  \
282  The cable works but is not 65W as advertised   
324                               Awesome Product   

                                        review_content  \
282  I have a pd supported car charger and I bought...   
324  Quick delivery.Awesome ProductPacking was good...   

                                              img_link  \
282  https://m.media-amazon.com/images/W/WEBP_40237...   
324  https://m.media-amazon.com/images/I/31-q0xhaTA...   

                                          product_link  
282  https://www.amazon.in/Amazon-Brand-Charging-Su...  
324  https://www.amazon.in/REDTECH-Lightning-Certif...

### Renomeando variáveis
Todas as variáveis serão renomeadas e postas em caixa-alta, para diferenciação dos registros e melhor entendimento do dataframe.

In [10]:
novas_cols = ["ID_PRODUTO", "NOME_PRODUTO", "CATEGORIA", "DESCONTO",
              "PRECO_REAL", "PORCENTAGEM_DESCONTO", "AVALIACAO",
              "TOTAL_AVALIACOES", "INFORMACOES", "ID_USUARIO", "USUARIO",
              "ID_REVISAO", "TITULO_REVISAO", "REVISAO", "URL_IMAGEM",
              "URL_PRODUTO"]

In [11]:
ajuste_var = {df.columns[i]:novas_cols[i] for i in range(len(df.columns))}
ajuste_var

{'product_id': 'ID_PRODUTO',
 'product_name': 'NOME_PRODUTO',
 'category': 'CATEGORIA',
 'discounted_price': 'DESCONTO',
 'actual_price': 'PRECO_REAL',
 'discount_percentage': 'PORCENTAGEM_DESCONTO',
 'rating': 'AVALIACAO',
 'rating_count': 'TOTAL_AVALIACOES',
 'about_product': 'INFORMACOES',
 'user_id': 'ID_USUARIO',
 'user_name': 'USUARIO',
 'review_id': 'ID_REVISAO',
 'review_title': 'TITULO_REVISAO',
 'review_content': 'REVISAO',
 'img_link': 'URL_IMAGEM',
 'product_link': 'URL_PRODUTO'}

In [12]:
df.rename(columns=ajuste_var, inplace=True)

In [13]:
df.columns

Index(['ID_PRODUTO', 'NOME_PRODUTO', 'CATEGORIA', 'DESCONTO', 'PRECO_REAL',
       'PORCENTAGEM_DESCONTO', 'AVALIACAO', 'TOTAL_AVALIACOES', 'INFORMACOES',
       'ID_USUARIO', 'USUARIO', 'ID_REVISAO', 'TITULO_REVISAO', 'REVISAO',
       'URL_IMAGEM', 'URL_PRODUTO'],
      dtype='object')

## Transformação de Dados

### Remoção de caracteres

In [14]:
df.head()

ID_PRODUTO                                       NOME_PRODUTO  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                           CATEGORIA DESCONTO PRECO_REAL  \
0  Computers&Accessories|Accessories&Peripherals|...     ₹399     ₹1,099   
1  Computers&Accessories|Accessories&Peripherals|...     ₹199       ₹349   
2  Computers&Accessories|Accessories&Peripherals|...     ₹199     ₹1,899   
3  Computers&Accessories|Accessories&Peripherals|...     ₹329       ₹699   
4  Computers&Accessories|Accessories&Peripherals|...     ₹154       ₹399   

  PORCENTAGEM_DESCONTO AVALIACAO TOTAL_AVALIACOES  \
0                  64%       4.2           24,269   
1                  43%       4.0           43,994   
2                  90%       3.9            7,928   
3                  53%       4.2           94,363   
4                  61%       4.2           16,905   

                                         INFORMACOES  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   

                                          ID_USUARIO  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   

                                             USUARIO  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   

                                          ID_REVISAO  \
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...   
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...   
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...   
3  R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...   
4  R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...   

                                      TITULO_REVISAO  \
0  Satisfied,Charging is really fast,Value for mo...   
1  A Good Braided Cable for Your Type C Device,Go...   
2  Good speed for earlier versions,Good Product,W...   
3  Good product,Good one,Nice,Really nice product...   
4  As good as original,Decent,Good one for second...   

                                             REVISAO  \
0  Looks durable Charging is fine tooNo complains...   
1  I ordered this cable to connect my phone to An...   
2  Not quite durable and sturdy,https://m.media-a...   
3  Good product,long wire,Charges good,Nice,I bou...   
4  Bought this instead of original apple, does th...   

                                          URL_IMAGEM  \
0  https://m.media-amazon.com/images/W/WEBP_40237...   
1  https://m.media-amazon.com/images/W/WEBP_40237...   
2  https://m.media-amazon.com/images/W/WEBP_40237...   
3  https://m.media-amazon.com/images/I/41V5FtEWPk...   
4  https://m.media-amazon.com/images/W/WEBP_40237...   

                                         URL_PRODUTO  
0  https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  
1  https://www.amazon.in/Ambrane-Unbreakable-Char...  
2  https://www.amazon.in/Sounce-iPhone-Charging-C...  
3  https://www.amazon.in/Deuce-300-Resistant-Tang...  
4  https://www.amazon.in/Portronics-Konnect-POR-1...

Ao olharmos para o conjunto de dados, percebemos alguns caracteres especiais. Sendo assim, faz-se necessário remover estes símbolos, presentes nas variáveis `DESCONTO`, `PRECO_REAL`, `PORCENTAGEM_DESCONTO` e `TOTAL_AVALIACOES`, para que seja possível efetuar operações matemáticas com as mesmas.

In [15]:
def remove_char(i, *args): # função que remove determinado caracter
    for arg in args:
        i = i.replace(arg, '')
    
    return i

In [16]:
df["DESCONTO"] = df.DESCONTO.apply(remove_char, args=('₹', ','))

In [17]:
df["PRECO_REAL"] = df.PRECO_REAL.apply(remove_char, args=('₹', ','))

In [18]:
df["PORCENTAGEM_DESCONTO"] = df.PORCENTAGEM_DESCONTO.apply(remove_char, args=('%'))

In [19]:
df["TOTAL_AVALIACOES"] = df.TOTAL_AVALIACOES.apply(remove_char, args=(','))

### Modificando dados
Os valores da variável `PORCENTAGEM_DESCONTO` serão passados para a forma decimal, a fim de representar porcentagem na forma decimal.

In [20]:
df["PORCENTAGEM_DESCONTO"] = df.PORCENTAGEM_DESCONTO.apply(lambda i: int(i)/100)

Será necessário remover a especificidade dos itens da variável `CATEGORIA`, fazendo uma generalização, para melhor entendimento e para futuras operações descritivas.

In [21]:
df["CATEGORIA"] = df.CATEGORIA.apply(lambda x: x.split('|')[0])

O tipo de algumas variáveis será modificado, para que possam ser realizadas operações matemáticas com as mesmas.

In [22]:
df.dtypes

ID_PRODUTO               object
NOME_PRODUTO             object
CATEGORIA                object
DESCONTO                 object
PRECO_REAL               object
PORCENTAGEM_DESCONTO    float64
AVALIACAO                object
TOTAL_AVALIACOES         object
INFORMACOES              object
ID_USUARIO               object
USUARIO                  object
ID_REVISAO               object
TITULO_REVISAO           object
REVISAO                  object
URL_IMAGEM               object
URL_PRODUTO              object
dtype: object

In [23]:
df["DESCONTO"] = df.DESCONTO.astype("float32") # número com casa decimal

In [24]:
df["PRECO_REAL"] = df.PRECO_REAL.astype("float32")

In [25]:
df["PORCENTAGEM_DESCONTO"] = df.PORCENTAGEM_DESCONTO.astype("float32")

In [26]:
df["TOTAL_AVALIACOES"] = df.TOTAL_AVALIACOES.astype("int32") # número inteiro

In [27]:
df.dtypes

ID_PRODUTO               object
NOME_PRODUTO             object
CATEGORIA                object
DESCONTO                float32
PRECO_REAL              float32
PORCENTAGEM_DESCONTO    float32
AVALIACAO                object
TOTAL_AVALIACOES          int32
INFORMACOES              object
ID_USUARIO               object
USUARIO                  object
ID_REVISAO               object
TITULO_REVISAO           object
REVISAO                  object
URL_IMAGEM               object
URL_PRODUTO              object
dtype: object

In [28]:
df

ID_PRODUTO                                       NOME_PRODUTO  \
0     B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1     B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2     B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3     B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4     B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   
...          ...                                                ...   
1460  B08L7J3T31  Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...   
1461  B01M6453MB  Prestige Delight PRWO Electric Rice Cooker (1 ...   
1462  B009P2LIL4  Bajaj Majesty RX10 2000 Watts Heat Convector R...   
1463  B00J5DYCCA  Havells Ventil Air DSP 230mm Exhaust Fan (Pist...   
1464  B01486F4G6  Borosil Jumbo 1000-Watt Grill Sandwich Maker (...   

                  CATEGORIA  DESCONTO  PRECO_REAL  PORCENTAGEM_DESCONTO  \
0     Computers&Accessories     399.0      1099.0                  0.64   
1     Computers&Accessories     199.0       349.0                  0.43   
2     Computers&Accessories     199.0      1899.0                  0.90   
3     Computers&Accessories     329.0       699.0                  0.53   
4     Computers&Accessories     154.0       399.0                  0.61   
...                     ...       ...         ...                   ...   
1460           Home&Kitchen     379.0       919.0                  0.59   
1461           Home&Kitchen    2280.0      3045.0                  0.25   
1462           Home&Kitchen    2219.0      3080.0                  0.28   
1463           Home&Kitchen    1399.0      1890.0                  0.26   
1464           Home&Kitchen    2863.0      3690.0                  0.22   

     AVALIACAO  TOTAL_AVALIACOES  \
0          4.2             24269   
1          4.0             43994   
2          3.9              7928   
3          4.2             94363   
4          4.2             16905   
...        ...               ...   
1460         4              1090   
1461       4.1              4118   
1462       3.6               468   
1463         4              8031   
1464       4.3              6987   

                                            INFORMACOES  \
0     High Compatibility : Compatible With iPhone 12...   
1     Compatible with all Type C enabled devices, be...   
2     【 Fast Charger& Data Sync】-With built-in safet...   
3     The boAt Deuce USB 300 2 in 1 cable is compati...   
4     [CHARGE & SYNC FUNCTION]- This cable comes wit...   
...                                                 ...   
1460  SUPREME QUALITY 90 GRAM 3 LAYER THIK PP SPUN F...   
1461                       230 Volts, 400 watts, 1 Year   
1462  International design and styling|Two heat sett...   
1463  Fan sweep area: 230 MM ; Noise level: (40 - 45...   
1464  Brand-Borosil, Specification – 23V ~ 5Hz;1 W...   

                                             ID_USUARIO  \
0     AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1     AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2     AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3     AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4     AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   
...                                                 ...   
1460  AHITFY6AHALOFOHOZEOC6XBP4FEA,AFRABBODZJZQB6Z4U...   
1461  AFG5FM3NEMOL6BNFRV2NK5FNJCHQ,AGEINTRN6Z563RMLH...   
1462  AGVPWCMAHYQWJOQKMUJN4DW3KM5Q,AF4Q3E66MY4SR7YQZ...   
1463  AF2JQCLSCY3QJATWUNNHUSVUPNQQ,AFDMLUXC5LS5RXDJS...   
1464  AFGW5PT3R6ZAVQR4Y5MWVAKBZAYA,AG7QNJ2SCS5VS5VYY...   

                                                USUARIO  \
0     Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1     ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2     Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3     Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4     rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   
...                                            

# ESTATÍSTICA DESCRITIVA
Com o dataframe tratado, será feita a organização, descrição e representação dos dados, através de métodos estatísticos descritivos.

## Distribuição de Frequências
É importante saber sobre a quantidade de ocorrências de determinados valores no conjunto de dados analisado. Para isso, é possível utilizar a distribuição de frequências.

Ao analisarmos a variável `CATEGORIA`, é possível ter uma noção sobre a demanda a determinado seguimento.

In [29]:
import plotly.express as px

In [30]:
tema = px.colors.qualitative.Light24

In [31]:
cont_categoria = px.histogram(df, x="CATEGORIA", width=900, text_auto=True,
                              color_discrete_sequence=tema, color="CATEGORIA")

cont_categoria.update_layout(font_size=12, xaxis=dict(title="Categoria"),
                             yaxis=dict(title="Contagem"),
                             title=dict(text="CONTAGEM DE CATEGORIAS",
                                        x=0.5, font_size=22),
                             plot_bgcolor="rgb(250, 250, 250)",
                             showlegend=False)

cont_categoria.update_xaxes(showline=True, linewidth=1, linecolor="lightgrey",
                            categoryorder="total descending")

cont_categoria.update_yaxes(showline=True, linewidth=1, linecolor="lightgrey")

Com isso, percebesse uma quantidade de amostras polarizada.  

A quantidade de avaliações acerca de determinado produto demonstra o interesse do consumidor em relatar algo sobre o mesmo.  Assim, se faz necessário uma análise acerca disso.

In [32]:
aval_categoria = px.strip(df, x="CATEGORIA", y="TOTAL_AVALIACOES", width=900,
                          color_discrete_sequence=tema, color="CATEGORIA")

aval_categoria.update_layout(font_size=12, xaxis=dict(title="Categoria"),
                             yaxis=dict(title="Contagem"),
                             title=dict(text="TOTAL DE AVALIAÇÕES POR CATEGORIA",
                                        x=0.5, font_size=22),
                             plot_bgcolor="rgb(250, 250, 250)",
                             showlegend=False)

aval_categoria.update_xaxes(showline=True, linewidth=1, linecolor="lightgrey")

aval_categoria.update_yaxes(showline=True, linewidth=1, linecolor="lightgrey")

Ao olharmos para o gráfico, é possível perceber muitos comentários sobre determinados produtos de algumas categorias e alguns despontam em relação a outros da mesma categoria.

## Medidas de Tendência Central
Uma representação da concentração dos valores de determinada variável pode ser obtida com as medidas de tendência central.  

Ao se analisar o percentual de desconto nas 4 categorias com maiores volumes de dados, utilizando a técnica do box plot, é possível verificar como os dados do percentual descontado estão distribuídos.

In [33]:
plot_box = df[df.CATEGORIA.isin(df.CATEGORIA.value_counts().index[:4])]

In [34]:
perc_desc_cat = px.box(plot_box, y="PORCENTAGEM_DESCONTO", width=900,
                       color_discrete_sequence=tema, color="CATEGORIA",
                       notched=True)

perc_desc_cat.update_layout(font_size=12, plot_bgcolor="rgb(250, 250, 250)",
                            title=dict(text="PERCENTUAL DE DESCONTO POR CATEGORIA",
                                       x=0.5, font_size=22),
                            legend=dict(title="Categoria", bgcolor="#F9F9F9",
                                        bordercolor="#D0D1D3",
                                        borderwidth=1,
                                        font=dict(size=14)))

perc_desc_cat.update_xaxes(showline=True, linewidth=1, linecolor="lightgrey")

perc_desc_cat.update_yaxes(showline=True, linewidth=1, linecolor="lightgrey",
                           title="Percentual Descontado")

Ao se analisar o gráfico, nota-se que a centralidade dos dados para as categorias *Computers&Accessories* e *Eletronics* gira em torno de 50%. Olhando para a categoria *OfficeProducts*, percebesse um percentual de desconto baixo, se comparado ao das demais categorias, porém, alguns de seus valores despontaram como outliers, ou seja, valores fora da curva, assumindo até 75% de desconto.

# ESTATÍSTICA INFERENCIAL
A partir dos resultados apresentados, serão feitas análises, com o propósito de se obter o raciocínio necessário para a partir dos dados, se obter uma conclusão geral.

## Correlação Linear
Através da correlação linear, obtém-se relações estatísticas entre as variáveis, úteis para  determinar (mensurar) o grau de relacionamento entre duas variáveis.

In [35]:
mapa_calor = px.imshow(df.corr(method="pearson"), text_auto=True)

mapa_calor.update_layout(width=900, plot_bgcolor="rgb(250, 250, 250)",
                         title=dict(text="CORRELAÇÃO ENTRE VARIÁVEIS NUMÉRICAS",
                                    x=0.5, font_size=22))

mapa_calor.update_xaxes(showline=True, linewidth=1, linecolor="lightgrey")

mapa_calor.update_yaxes(showline=True, linewidth=1, linecolor="lightgrey")

Ao olhar para o gráfico de calor, é possível perceber uma correlação positiva muito forte entre as variáveis `PRECO_REAL` e `DESCONTO`. Essa correlação pode ser visualizada através de um gráfico de dispersão.

In [36]:
dispersao_precos = px.scatter(df, x="DESCONTO", y="PRECO_REAL", trendline="ols")

dispersao_precos.update_layout(font_size=12, xaxis=dict(title="Desconto"),
                               width=900, yaxis=dict(title="Preços"),
                               title=dict(text="CORRELAÇÃO ENTRE PREÇO REAL E DESCONTO",
                                          x=0.5, font_size=22),
                               plot_bgcolor="rgb(250, 250, 250)",
                               showlegend=False)

dispersao_precos.update_xaxes(showline=True, linewidth=1, linecolor="lightgrey")

dispersao_precos.update_yaxes(showline=True, linewidth=1, linecolor="lightgrey")

É possível confirmar uma correlação linear positiva em relação as variáveis PRECO_REAL e DESCONTO, ou seja, quando uma aumenta a outra também aumenta. Com isso, se entende que essas duas variáveis estão intimamente ligadas.

## Perguntas de Negócio
As seguintes perguntas de negócio serão respondidas:
* Um item da categoria OfficeProducts tem quantos por cento de chance de possuir mais de 20% de desconto?
* Qual a chance de um produto da categoria Computers&Accessories não ter mais de 50% de desconto?
* Adquirindo um produto da categoria Electronics que custa mais ou ₹1000 (mil rúpias), qual a probabilidade de ele possuir mais de 60% de desconto?
* Ao se comprar 10 produtos da categoria Computers&Accessories, qual a chance de pelo menos um possuir um desconto maior ou igual a 80%?
* Comprando-se 5 produtos da categoria Electronics, qual a probabilidade de nenhum possuir um desconto superior ou igual a 50%?

**1) Qual a probabilidade de um item da categoria OfficeProducts possuir pelo menos 20% de desconto?**

In [37]:
prob = lambda A, E: round((A / E) * 100, 2) if A < E else False

In [38]:
pergunta_1E = df[df.CATEGORIA.isin(["OfficeProducts"])]
pergunta_1A = pergunta_1E.query("PORCENTAGEM_DESCONTO > 0.20")

In [39]:
prob(len(pergunta_1A), len(pergunta_1E))

19.35

A probabilidade de um item da categoria *OfficeProducts* possuir mais de 20% de desconto é de 19,35%.
<br><br>
**2) Qual a chance de um produto da categoria Computers&Accessories não ter mais de 50% de desconto?**

In [40]:
prob_nao = lambda A, E: round((1 - (A / E)) * 100, 2) if A < E else False

In [41]:
pergunta_2E = df[df.CATEGORIA.isin(["Computers&Accessories"])]
pergunta_2A = pergunta_2E.query("PORCENTAGEM_DESCONTO > 0.50")

In [42]:
prob_nao(len(pergunta_2A), len(pergunta_2E))

37.09

A chance de um produto da categoria *Computers&Accessories* não ter 50% de desconto (ou mais) é de 37.09%, o que torna uma chance relativamente moderada de possuir desconto superior a faixa indicada. Com isso, se deduz que produtos desta categoria geralmente possuem bons descontos.
<br><br>
**3) Adquirindo um produto da categoria Electronics que custa mais ou ₹1000 (mil rúpias), qual a probabilidade de ele possuir mais de 60% de desconto?**

In [43]:
pergunta_3E = df.loc[df.CATEGORIA == "Electronics"].query("PRECO_REAL > 1000")
pergunta_3A = pergunta_3E.query("PORCENTAGEM_DESCONTO > 0.60")

In [44]:
prob(len(pergunta_3A), len(pergunta_3E))

36.5

Existe 36.5% de chance de um produto que custa mais de ₹1000 possuir mais de 60% de desconto.
<br><br>
**4) Ao se comprar 10 produtos da categoria Computers&Accessories, qual a chance de pelo menos um possuir um desconto maior ou igual a 80%?**

In [45]:
from scipy.stats import binom

In [46]:
pergunta_4E = pergunta_2E.copy()
pergunta_4A = pergunta_4E.loc[pergunta_4E.PORCENTAGEM_DESCONTO >= 0.80]
p4 = prob(len(pergunta_4A), len(pergunta_4E)) / 100

In [47]:
round((1 - binom.pmf(0, 10, p4)) * 100, 2)

62.2

A probabilidade de pelo menos um produto da categoria Computers&Accessories possuir um desconto maior que 70% é de 62,2%.
<br><br>
**5) Comprando-se 5 produtos da categoria Electronics, qual a probabilidade de nenhum possuir um desconto superior ou igual a 50%?**  

Para esta operação utilizaremos a função geom.pmf da biblioteca scipy, porém, será feito um pequeno ajuste, para ela retorna em forma de porcentagem arredondada.

In [48]:
from scipy.stats import geom

In [49]:
geom_pmf = lambda n, p: round(geom.pmf(n, p) * 100, 2)

In [50]:
pergunta_5E = df[df.CATEGORIA.isin(["Electronics"])]
pergunta_5A = pergunta_5E.query("PORCENTAGEM_DESCONTO >= 0.50")
p5 = prob(len(pergunta_5A), len(pergunta_5E)) / 100

In [51]:
geom_pmf(6, p5)

0.9

As chances de tal prerrogativa ocorrer é baixíssima, não alcançando 1%. Assim, se entende que é quase certo que ao comprar um produto da categoria supracitada, ele terá um desconto igual ou superior a 50%.